docs/api/proposed_endpoints.md

```diff
@@ -0,0 +1,186 @@
+[TOC]
+
+# Posts related actions
+
+## Create a post
+
+Create a post of contents.
+
+```
+POST api/posts
+```
+
+**Paramaters**
+
+Name         | Description
+-------------|--------------------------------------
+id           | identity of the post
+posted_by    | user who is posting the content
+posted_at    | social media ploatform to post
+posted_time  | time at which post should be posted
+content_type | type of content, text or image
+content      | content of the post
+image        | image in the post
+is_approved  | whether the post is approved or not
+is_posted    | whether the post is posted or not
+
+**Request**
+```json
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "",
+    "content_type": "text",
+    "content": "Anything that user has to write",
+    "image": "",
+    "is_approved": "False",
+    "is_posted": "False"
+}
+```
+
+**Response**
+```json
+
+Status: 201 Created
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "",
+    "content_type": "text",
+    "content": "Anything that user has to write",
+    "image": "http://xyz.com/path/of/image.jpg",
+    "is_approved": "False",
+    "is_posted": "False"
+
+}
+```
+
+## Update a post
+
+```
+PATCH api/posts/:post_id (Requires Authorization)
+```
+
+**Parameters**
+
+Name         | Description
+-------------|--------------------------------------
+new_content  | new content of the post
+new_image    | new image in the post
+posted_time  | time at which post should be posted
+content_type | current content type of the post
+
+**Request**
+```json
+{
+    "new_content": "",
```

Comment:
The key is always same.

You are updating content doesn't mean you say `new_content`.

In `PATCH` request, user is free to update any part of the resource.

Reply:
True that. Uniform attribute names are easier to understand. Since the context is same here as well it's better to use `content` and `image` just like you did with `content_type` and `posted_time`.